data-raw/simulate_longitudinal_data.R

```diff
@@ -1,320 +1,72 @@
-    Mapped_COUNTRY = mapped$Mapped_COUNTRY,
-    lWorkflows = kri_wf,
-    lAnalyzed = kris,
-    dSnapshotDate = endDate[i],
-    strStudyID = "AA-AA-000-0000"
-  )
-  reporting_wf_site <- MakeWorkflowList(strPath = "workflow/3_reporting")
-  reporting_site <- RunWorkflows(reporting_wf_site, lReporting_Input_site)
-
-  lReporting_Input_country <- list(
-    Raw_ctms_site = clindata::ctms_site,
-    Raw_ctms_study = clindata::ctms_study,
-    Mapped_ENROLL = mapped$Mapped_ENROLL,
-    Mapped_STUDY = mapped$Mapped_STUDY,
-    Mapped_SITE = mapped$Mapped_SITE,
-    Mapped_COUNTRY = mapped$Mapped_COUNTRY,
-    lWorkflows = cou_wf,
-    lAnalyzed = cous,
-    dSnapshotDate = endDate[i],
-    strStudyID = "AA-AA-000-0000"
-  )
-  reporting_wf_country <- MakeWorkflowList(strPath = "workflow/3_reporting")
-  reporting_country <- RunWorkflows(reporting_wf_country, lReporting_Input_country)
-
-  if (i == 1) {
-    lAnalysis_site <- kris
-    lAnalysis_country <- cous
-    lReporting_site <- reporting_site
-    lReporting_country <- reporting_country
-  }
-  else {
-  lAnalysis_site <- imap(lAnalysis_site, \(this, metric)
-                         imap(this[grep("Analysis", names(this))], \(x, idx)
-                              bind_rows(x, kris[[metric]][[idx]])))
-  lAnalysis_country <-  imap(lAnalysis_site, \(this, metric)
-                             imap(this[grep("Analysis", names(this))],\(x, idx)
-                                  bind_rows(x, cous[[metric]][[idx]])))
-
-  lReporting_site <- imap(lReporting_site[grep("Reporting", names(lReporting_site))], \(x, idx)
-                               bind_rows(x, reporting_site[[idx]]))
-  lReporting_country <-  imap(lReporting_country[grep("Reporting", names(lReporting_country))], \(x, idx)
-                              bind_rows(x, reporting_country[[idx]]))
-  }
-}
-
-#only save the most recent groups, bounds and metrics entry for use in reporting
-lReporting_site$Reporting_Groups <- reporting_site$Reporting_Groups
-lReporting_site$Reporting_Bounds <- reporting_site$Reporting_Bounds
-lReporting_site$Reporting_Metrics <- reporting_site$Reporting_Metrics %>%
-  filter(str_detect(MetricID, "Analysis_kri"))
-
-lReporting_country$Reporting_Groups <- reporting_country$Reporting_Groups
-lReporting_country$Reporting_Bounds <- reporting_country$Reporting_Bounds
-lReporting_country$Reporting_Metrics <- reporting_country$Reporting_Metrics %>%
-  filter(str_detect(MetricID, "Analysis_cou"))
+core_mappings <- c("AE", "COUNTRY", "DATACHG", "DATAENT", "ENROLL", "LB",
+                   "PD", "PK", "QUERY", "STUDY", "STUDCOMP", "SDRGCOMP", "SITE", "SUBJ")
+
+basic_sim <- gsm.datasim::generate_rawdata_for_single_study(
+  SnapshotCount = 12,
+  SnapshotWidth = "months",
+  ParticipantCount = 1000,
+  SiteCount = 150,
+  StudyID = "AA-AA-000-0000",
+  workflow_path = "workflow/1_mappings",
+  mappings = core_mappings,
+  package = "gsm.mapping",
+  desired_specs = NULL
+)
+
+lSource <- basic_sim[[12]]
```

Comment:
This causes `reportingXXX` objects to all have the same `SnapshotDate` and thereffore example 3.3 will not output anything